servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/H2ServerParentConnectionContext.java

```diff
@@ -128,13 +134,17 @@ protected void handleSubscribe(final Subscriber<? super H2ServerParentConnection
                     pipeline = channel.pipeline();
 
                     parentChannelInitializer = new DefaultH2ServerParentConnection(connection, subscriber,
-                            delayedCancellable, NettyPipelineSslUtils.isSslEnabled(pipeline));
+                            delayedCancellable, NettyPipelineSslUtils.isSslEnabled(pipeline),
+                            connectionObserver(channel));
 
                     new H2ServerParentChannelInitializer(h2ServerConfig,
-                        new io.netty.channel.ChannelInitializer<Channel>() {
+                        new io.netty.channel.ChannelInitializer<Http2StreamChannel>() {
                             @Override
-                            protected void initChannel(final Channel streamChannel) {
+                            protected void initChannel(final Http2StreamChannel streamChannel) {
                                 connection.trackActiveStream(streamChannel);
+                                final StreamObserver streamObserver = registerStreamObserver(streamChannel,
```

Comment:
If this throws, do we cleanup and propagate async control flow?

Reply:
@NiteshKant this scenario should be visible via `ConnectionObserver.connectionClosed(Throwable)` because netty will propagate this exception through the pipeline.

@Scottmitch added try-catch for an observer invocation.

Reply:
For the connection, time delta between `onNewConnection()` and `established*()` is significant because it involves a TCP syn-ack and optionally an SSL handshake. 

Stream creation happens locally without network interaction. Therefore, the time between a request for a new stream and `childChannelActive` is negligible. Can not imagine a use-case when having 2 events will provide value for users.